src/Uno.UWP/Storage/Streams/DataWriter.cs

```diff
@@ -0,0 +1,287 @@
+
+		/// <summary>
+		/// Writes a floating-point value to the output stream.
+		/// </summary>
+		/// <param name="value">The value.</param>
+		public void WriteSingle(float value) => WriteChunkToUnstoredBuffer(BitConverter.GetBytes(value));
+
+		/// <summary>
+		/// Writes a floating-point value to the output stream.
+		/// </summary>
+		/// <param name="value">The value.</param>
+		public void WriteDouble(double value) => WriteChunkToUnstoredBuffer(BitConverter.GetBytes(value));
+
+		/// <summary>
+		/// Writes a date and time value to the output stream.
+		/// </summary>
+		/// <param name="value">The value.</param>
+		public void WriteDateTime( DateTimeOffset value)
+		{ 
+			// Implementation in UWP is using Win32 FileTime format (starting on 1st January 1601)
+			// As contrary to Win32 however also supports pre-1601 dates
+			var epoch1601 = new DateTimeOffset(1601, 1, 1, 0, 0, 0, TimeSpan.Zero);
+			long ticks = (value - epoch1601).Ticks;
+			WriteInt64(ticks);
+		}
+
+		/// <summary>
+		/// Writes a time-interval value to the output stream.
+		/// </summary>
+		/// <param name="value">The value.</param>
+		public void WriteTimeSpan(TimeSpan value) => WriteChunkToUnstoredBuffer(BitConverter.GetBytes(value.Ticks));
+
+		/// <summary>
+		/// Writes a string value to the output stream.
+		/// </summary>
+		/// <param name="value">The value.</param>
+		/// <returns>The length of the string, in bytes.</returns>
+		public uint WriteString(string value)
+		{
+			byte[] stringBytes;
+			switch (UnicodeEncoding)
+			{
+				case UnicodeEncoding.Utf8:
+					stringBytes = Encoding.UTF8.GetBytes(value);
+					break;
+				case UnicodeEncoding.Utf16LE:
+					stringBytes = Encoding.Unicode.GetBytes(value);
+					break;
+				case UnicodeEncoding.Utf16BE:
+					stringBytes = Encoding.BigEndianUnicode.GetBytes(value);
+					break;
+				default:
+					throw new InvalidOperationException("Unsupported UnicodeEncoding value.");
+			}
+			_memoryStream.Write(stringBytes, 0, stringBytes.Length);
+			return (uint)stringBytes.Length;
+		}
+
+		/// <summary>
+		/// Gets the size of a string.
+		/// </summary>
+		/// <param name="value">The string.</param>
+		/// <returns>The size of the string, in bytes.</returns>
+		public uint MeasureString(string value)
+		{
+			switch (UnicodeEncoding)
+			{
+				case UnicodeEncoding.Utf8:
+					return (uint)Encoding.UTF8.GetByteCount(value);
+				case UnicodeEncoding.Utf16LE:
+					return (uint)Encoding.Unicode.GetByteCount(value);
+				case UnicodeEncoding.Utf16BE:
+					return (uint)Encoding.BigEndianUnicode.GetByteCount(value);
+				default:
+					throw new InvalidOperationException("Unsupported UnicodeEncoding value.");
+			}
+		}
+
+		/// <summary>
+		/// Detaches the buffer that is associated with the data writer.
+		/// </summary>
+		/// <returns>The detached buffer.</returns>
+		public IBuffer DetachBuffer()
+		{
+			var result = new Buffer(_memoryStream.ToArray());
+			_memoryStream.Dispose();
```

Comment:
Codacy found an issue: [Move this 'Dispose' call into this class' own 'Dispose' method.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)